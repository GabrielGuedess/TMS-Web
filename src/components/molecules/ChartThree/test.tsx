import { type ReactNode } from 'react';

import { render } from '@testing-library/react';

import { ChartThree } from '.';

jest.mock('components/atoms/ApexChart', () => ({
  ApexChart: ({ children }: { children: ReactNode }) => (
    <div data-testid="Mock ApexChart">{children}</div>
  ),
}));

describe('<ChartThree />', () => {
  it('should render the chart three correctly', () => {
    const { container } = render(<ChartThree />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
