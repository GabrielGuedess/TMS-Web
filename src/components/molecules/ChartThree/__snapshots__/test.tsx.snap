// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChartThree /> should render the chart three correctly 1`] = `
<div
  class="col-span-12 rounded border bg-white-lilac-50 px-5 pb-5 pt-[1.875rem] shadow-default transition-all dark:border-shark-950 dark:bg-smoke-950 sm:px-[1.875rem] xl:col-span-5"
>
  <div
    class="mb-3 justify-between gap-4 sm:flex"
  >
    <div>
      <h5
        class="text-xl font-semibold text-shark-900 dark:text-white-lilac-50"
      >
        An√°lise de Visitantes
      </h5>
    </div>
    <div>
      <div
        class="relative z-20 inline-block"
      >
        <select
          class="relative z-20 inline-flex appearance-none bg-transparent py-1 pl-3 pr-9 text-sm font-medium outline-none"
        >
          <option
            value=""
          >
            Mensal
          </option>
          <option
            value=""
          >
            Anual
          </option>
        </select>
        <span
          class="absolute right-3 top-1/2 z-10 -translate-y-1/2"
        >
          <svg
            class="text-comet-500"
            fill="none"
            height="18"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19.9201 8.95L13.4001 15.47C12.6301 16.24 11.3701 16.24 10.6001 15.47L4.08008 8.95"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-miterlimit="10"
              stroke-width="1.5"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
  <div
    class="flex size-full items-center justify-center"
  >
    <div
      data-testid="Mock ApexChart"
    />
  </div>
</div>
`;
